fedimint-client/src/transaction/builder.rs

```diff
@@ -475,44 +481,36 @@ impl TransactionBuilder {
         let input_states = self
             .inputs
             .into_iter()
+            .filter(|bundle| !bundle.is_empty())
```

Comment:
Core of the change: filter out "empty bundles" explicitly upfront, so we can simplify rest of the code.